python/cudf/cudf/core/dataframe.py

```diff
@@ -104,6 +103,13 @@
     _external_only_api,
 )
 
+if sys.version_info >= (3, 10):
+    # Introduced in Python 3.10
+    from types import NotImplementedType
+else:
+    NotImplementedType = Any
```

Comment:
I'm a bit surprised that this logic works despite https://github.com/rapidsai/cudf/pull/11640#discussion_r961524007. I guess the parsing of the Python version from `sys.version_info` is something that mypy does support despite the static nature of the analysis?

Reply:
https://mypy.readthedocs.io/en/stable/common_issues.html#python-version-and-system-platform-checks

Reply:
Nice that's cool.